tests/integration_tests/assets/enable_coverage.py

```diff
@@ -0,0 +1,28 @@
+from pathlib import Path
+
+services = [
+    "cloud-init-local.service",
+    "cloud-init.service",
+    "cloud-config.service",
+    "cloud-final.service",
+]
+service_dir = Path("/lib/systemd/system/")
+
+# Check for the existence of the service files
+for service in services:
+    if not (service_dir / service).is_file():
+        print(f"Error: {service} does not exist in {service_dir}")
+        exit(1)
+
+# Prepend the ExecStart= line with 'python3 -m coverage run'
+for service in services:
```

Comment:
I've done similar things (more ad-hoc) for inserting profiler commands into the service files. At some point I'll probably steal this code or make it more generic to do that in automatically in tox (unless someone else wants to do it)

Reply:
https://github.com/canonical/cloud-init/issues/4729